### PR TITLE
[GTW-7065] Automated retries for SCA soft declines

### PR DESCRIPTION
**Pull request by mathieu-carlier-cko**
# Description of changes in PR

[//]: # 'Please add your description here'
Update documentation for automated retries on SCA soft declines. Adding the new properties on the request & response.

## Checklist

- [x] Reviewed by the Gateway product team
- [x] Added a changelog entry to the `changelog.md` file in either `abc_spec` or `nas_spec`
- [x] Contacted the Tech Docs team about any corresponding guides that need to be updated for www.checkout.com/docs

**Contributing options**: Look at [our documentation](https://checkout.atlassian.net/wiki/spaces/PD/pages/2169506663/API+ref+publication+process) for options to contribute to the API reference.

### CODE REVIEW

File: nas_spec/components/schemas/Payments/3dsEnrollmentData.yaml [hunk 1]
```diff
@@ -12,3 +12,7 @@ properties:
         * `N` - Customer not enrolled
         * `U` - Unknown
     example: Y
+  upgrade_reason:
+    type: string
+    description: Indicates the reason why the payment was upgraded to a 3D secure payment (when `allow_upgrade` is specified)
```
Comment (alex-wernick-cko) [inline nas_spec/components/schemas/Payments/3dsEnrollmentData.yaml:17]: Wording can be simplified a tiny bit
```suggestion
    description: Indicates the reason why the payment was upgraded to 3D secure (when `allow_upgrade` is specified)
```

File: nas_spec/components/schemas/Payments/3dsData.yaml [hunk 2]
```diff
@@ -55,3 +55,7 @@ properties:
     type: boolean
     description: Indicates whether this session involved a challenge. This will only be set after communication with the scheme is finished.
     example: true
+  upgrade_reason:
+    type: string
+    description: Indicates the reason why the payment was upgraded to a 3D secure payment (when `allow_upgrade` is specified)
```
Comment (alex-wernick-cko) [inline nas_spec/components/schemas/Payments/3dsData.yaml:60]:
```suggestion
    description: Indicates the reason why the payment was upgraded to 3D secure (when `allow_upgrade` is specified)
```

File: abc_spec/components/schemas/Payments/3dsRequest.yaml [hunk 3]
```diff
@@ -51,3 +51,9 @@ properties:
       - challenge_requested
       - challenge_requested_mandate
     default: no_preference
+  allow_upgrade:
+    type: boolean
+    description: Whether to process this payment as a 3D Secure payment if the authorization was soft declined due to 3DS authentication required
```
Comment (alex-wernick-cko) [inline abc_spec/components/schemas/Payments/3dsRequest.yaml:56]:
```suggestion
    description: Whether to process this payment as 3D Secure if the authorization was soft declined due to 3DS authentication required
```

File: abc_spec/components/schemas/Payments/3dsEnrollmentData.yaml [hunk 4]
```diff
@@ -12,3 +12,7 @@ properties:
         * `N` - Customer not enrolled
         * `U` - Unknown
     example: Y
+  upgrade_reason:
+    type: string
+    description: Indicates the reason why the payment was upgraded to a 3D secure payment (when `allow_upgrade` is specified)
```
Comment (alex-wernick-cko) [inline abc_spec/components/schemas/Payments/3dsEnrollmentData.yaml:17]:
```suggestion
    description: Indicates the reason why the payment was upgraded to 3D secure (when `allow_upgrade` is specified)
```

File: abc_spec/components/schemas/Payments/3dsData.yaml [hunk 5]
```diff
@@ -41,3 +41,7 @@ properties:
     type: string
     description: Specifies an exemption reason so that the payment is not processed using 3D Secure authentication
     example: 'low_value'
+  upgrade_reason:
+    type: string
+    description: Indicates the reason why the payment was upgraded to a 3D secure payment (when `allow_upgrade` is specified)
```
Comment (alex-wernick-cko) [inline abc_spec/components/schemas/Payments/3dsData.yaml:46]:
```suggestion
    description: Indicates the reason why the payment was upgraded to 3D secure (when `allow_upgrade` is specified)
```

File: nas_spec/components/schemas/Payments/3dsRequest.yaml [hunk 6]
```diff
@@ -99,3 +99,8 @@ properties:
     maxLength: 1
     minLength: 1
     description: The algorithm used by the Issuer ACS to calculate the Authentication cryptogram. This is the AVALGO data returned in Cartes bancaires Authentication.
+  allow_upgrade:
+    type: boolean
+    description: Whether to process this payment as a 3D Secure payment if the authorization was soft declined due to 3DS authentication required
```
Comment (alex-wernick-cko) [inline nas_spec/components/schemas/Payments/3dsRequest.yaml:104]:
```suggestion
    description: Whether to process this payment as 3D Secure if the authorization was soft declined due to 3DS authentication required
```

**Review by alex-wernick-cko — APPROVED** [review]
Some suggestions. I don't feel that strongly but I think this working is a bit more concise. I'll approve anyway and leave it to you to decide if you want to take them or not

**Comment by sebastian-garcia-cardona-cko** [comment]
@mathieu-carlier-cko just a reminder that your branch has merge conflicts. You'll need to resolve these so reviewers can approve

**Review by Joshua-Bedeau-cko — APPROVED** [review]

**Review by evangelos-gkavogiannis-cko — DISMISSED** [review]

**Review by evangelos-gkavogiannis-cko — APPROVED** [review]

**Comment by Joshua-Bedeau-cko** [comment]
LGTM

**Review by Joshua-Bedeau-cko — APPROVED** [review]
LGTM

**Review by sebastian-garcia-cardona-cko — APPROVED** [review]

**Pull request merged**